h/services/search_index/__init__.py

```diff
@@ -1 +1,2 @@
-from h.services.search_index.service import factory  # noqa: F401
+from h.services.search_index._factory import factory  # noqa: F401
```

Comment:
The `_factory` isn't exactly as per our bounded context rules, but it just turns out to be very awkward to mock if you do `from ...factory import factory` as the patch doesn't know which factory you mean.

Reply:
I think this may be an issue with having a `factory.py` that contains a `factory()` and then trying to import that `factory()` into `__init__.py`, might want to avoid that. Options:

* Maybe the `factory()` function could go in `service.py` (although it looks like you wanted it out of there so that the test file wasn't so long?)
* `factory()` could be inlined directly in `__init__.py` (which would allow its tests to live in `__init___test.py`)
* Rename `factory.py` to `service_factory.py`
* Don't import `factory()` into `__init__.py`. Rename `_factory.py` to `factory.py`. Whereever `factory()` is referenced or imported it'll have to be referenced as `h.services.search_index.factory.factory`. The reason we import stuff into `__init__.py` is to make imports easier to write (and less likely to have to change it code is reorganised) but I'm guessing `factory()` is only imported in one or two places so not such a big deal in this case

Reply:
I think `service_factory.py` is probably the easiest thing to do here. I'll make the change.

I quite like the succinct nature of `h.services.search_index.factory`. It's just exactly what it is.